### ISSUE TITLE
Personal build information/help

### ISSUE THREAD

**Issue by debo** [coseyfannitutti/discipline#45]
Hi @coseyfannitutti I'm writing to you because, as per subject I'd like to build the Discipline v2 by myself but by being completely new to the hobby, I'm not quite sure where to go to get the parts manufactured, specifically the pcb/plates and the acrylic case.

If my understanding is correct, based on the online stream I saw with MechMerlin you usually use JLC PCB for prototyping and Elecrow for the final product, is that correct? In a normal scenario both the bottom and top plate are made of PCB material too? Just to understand if I can get them made all by Elecrow.

About the acrylic instead, I understand from the documentation that you use Ponoko instead. Is that correct too?

One more question, I notice in the photos that there are various engraving on the components are those part of the gerber and dfx files already or do I need to modify something else?

Last but not least, when I use JLC PCB/Elecrow, are there settings on their tool that I have to change other than the colour of the PCB?

Thanks a lot for your time and help.

**Comment by NiciTheNici** [comment]
Im currently in the process of building this keyboards as well, I feel like I understand the process well enough to answer some of your questions.

> you usually use JLC PCB for prototyping and Elecrow for the final product

Prototyping is not required unless you modify the PCB yourself and since coseyfannitutti's design has been thoroughly tested you can skip the Prototyping step all together and order straight from JLCPCB or Elecrow, although most people i know ordered from JLCPCB and even the guide recommended it.

> Both the bottom and top plate are made of PCB material too

You're absolutely right.

> I notice in the photos that there are various engraving on the components are those part of the gerber and dfx files already or do I need to modify something else?

You dont need to modify anything, your PCB will look just like in the photos if you just upload the gerber file as is. If you know how to you can add your own logos/markings to the PCB this is not necessary though.

> Are there settings on their tool that I have to change

Not really, you might wanna change the [surface finish](http://blog.optimumdesign.com/pcb-surface-finishes-comparison-hasl-osp-enig) but this is not necessary. I also recommend removing the order number, else it might be visible and that wouldnt be very pretty.

Heres my settings I used for comparison. I used the same settings for the top and bottom PCB.

![image](https://user-images.githubusercontent.com/51826257/117537563-30cb8700-b002-11eb-9f51-33d9a92831f5.png)

And heres how your shoppingcart should more or less look like.

![image](https://user-images.githubusercontent.com/51826257/117537660-c23af900-b002-11eb-9bcc-2b27accd6cc9.png)

If you have any more questions don't hesitate asking them! Good luck and have fun :)

**Comment by debo** [comment]
Thanks a lot @SuperNici this was super super helpful, really appreciated.

**Comment by robopuff** [comment]
Hey guys, I am building it for myself as well and I've used JLCPCB and it looks and works well. It was not easy for me to get proper M2 screws and standoffs since no store would sell one with the correct size and esthetically pleasing - aliexpress worked fine for it. With that, I've also ordered a steel plate and it is a blast to use! 

~Now I need to cut just the plexi and was wondering if you know the correct thickness @SuperNici for the kitfoot?~ scratch that, I've seen you've made a pr with it

**Comment by fourstepper** [comment]
@SuperNici would you care to elaborate how you handled the top plate? Looking at my order, the size doesn't get correctly detected by jlcpcb. Any ideas? Thank you a lot :)
<img width="967" alt="image" src="https://user-images.githubusercontent.com/49439044/140604302-5e8c06aa-3123-439f-b470-90b271cc6e03.png">

**Comment by robopuff** [comment]
You can get the dimensions of the top plate from laser cut files, can't remember what they are right now but I've used it to cut in steel plate and it fits perfectly

**Comment by fourstepper** [comment]
Sorry for my ignorance, I am still quite new to this. Which files are the laser cut files? Thank you a lot

**Comment by robopuff** [comment]
> Sorry for my ignorance, I am still quite new to this. Which files are the laser cut files? Thank you a lot

No worries, I had a million questions when I've built mine 😅

You can check it here https://github.com/coseyfannitutti/discipline/blob/master/plate/discipline-plate-lasercutting.dxf

**Comment by fourstepper** [comment]
Thank you, that worked!

**Comment by NiciTheNici** [comment]
Personally I made my discipline plateless, see [my reddit post ](https://www.reddit.com/r/MechanicalKeyboards/comments/qmvntb/comment/hjc6ruf/?utm_source=share&utm_medium=web2x&context=3). In hindsight I regret not having ordered plates, but it works just fine without one.

The file you imported was the Gerber file, the missing dimensions should be the same as for the PCB itself, in this case:

~~`122.81 * 304.89`~~ --> `305.7 x 96.2`

This is an issue with the generated Gerber file, I tried fixing it and had no luck. I'll open an issue in just a second.

PS. @fourstepper you just replied while I was writing this comment, i'll still post it if others have the same issue :)

**Comment by robopuff** [comment]
> Personally I made my discipline plateless, see [my reddit post ](https://www.reddit.com/r/MechanicalKeyboards/comments/qmvntb/comment/hjc6ruf/?utm_source=share&utm_medium=web2x&context=3). In hindsight I regret not having ordered plates, but it works just fine without one.
> 
> The file you imported was the Gerber file, the missing dimensions should be the same as for the PCB itself, in this case:
> 
> `122.81 * 304.89`
> 
> This is an issue with the generated Gerber file, I tried fixing it and had no luck. I'll open an issue in just a second.
> 
> PS. @fourstepper you just replied while I was writing this comment, i'll still post it if others have the same issue :)

Isn't PCB taller than the plate? I mean it's not covering the electronic components but rather just the keys

**Comment by NiciTheNici** [comment]
Oh I totally forgot about that, my bad. It would probably still arrive fine, you'd just pay more... (JLCPCB would just waste a bunch of extra Fr4 Material since the keyboard plate itself gets cut out of the specified size)

What measurements did @fourstepper use? I'll update my comment accordingly.

**Comment by fourstepper** [comment]
I used the measurements from the acrylic top plate as suggested by @robopuff, which reports 305.7 x 96.2 mm. Seems like that could work?

**Comment by NiciTheNici** [comment]
Thanks, that looks about right, I updated [the issue I opened](https://github.com/coseyfannitutti/discipline/issues/60) 

Thanks ( ´ ▽ ` )b

**Comment by fourstepper** [comment]
Thank you!

**Comment by robopuff** [comment]
> 
> 
> I used the measurements from the acrylic top plate as suggested by @robopuff, which reports 305.7 x 96.2 mm. Seems like that could work?

I've just checked against what service provider I've used sent back to me and indeed it was 305x96mm JLCPCB have their checks done before accepting a job so maybe they would take a look into it as well before misprinting/cutting/etc things 😅 

btw order standoffs and screws ahead of time from whatever source, I've waited quite long before they were shipped from China 😞

**Comment by fourstepper** [comment]
@robopuff I have a lot of time and have some screws and standoffs already, as I am also in the process of building a CRKBD. Thank you for the recommendation though!

**Comment by NiciTheNici** [comment]
> I've just checked against what service provider I've used sent back to me and indeed it was 305x96mm JLCPCB have their checks done before accepting a job so maybe they would take a look into it as well before misprinting/cutting/etc things 😅



@robopuff i dont think it should matter too much as long as it is big enough, the exact cutting measurements are taken from the gerber file after all so it really should not be a problem

**Comment by NiciTheNici** [comment]
Heyall, just out of curiosity... How many of you ended up building this keyboard? Got any pics?

Heres mine!
![image](https://github.com/coseyfannitutti/discipline/assets/51826257/ffa788b1-0ee6-4d5b-975b-a5824eebded1)

**Comment by robopuff** [comment]
Daily driven since built, use U4T switches with steel plate, it's a joy to use. 

![PXL_20230527_203731344.jpg](https://github.com/coseyfannitutti/discipline/assets/730047/729e58f9-cebf-4cc7-98fb-3c633c6e033e)

**Comment by rossw42** [comment]
What thickness do folks laser cut the plate? In one spot - the acrylic case build up it says 1.5mm for the plate?
